### PR TITLE
Fix prefetched dehydrated state to support Dates, Maps, etc (patch)

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes: ?

### What are the changes and their implications?

`dehydratedState` will now be passed through SuperJSON, adding support for `Date`, `Map`, `Set`, ...

## Bug Checklist

- [x] Integration test added (see [test docs](https://blitzjs.com/docs/contributing#running-tests) if needed)

### CODE REVIEW

File: nextjs/run-tests.js [hunk 1]
```diff
@@ -65,7 +65,9 @@ async function main() {
   }
 
   console.log('Running tests with concurrency:', concurrency)
-  let tests = process.argv.filter((arg) => arg.endsWith('.test.js'))
+  let tests = process.argv.filter(
+    (arg) => arg.endsWith('.test.js') || arg.endsWith('.test.ts')
```
Comment (Skn0tt) [inline nextjs/run-tests.js:69]: @flybayer could this become a problem when merging Next.js upstream? Should I contribute this to Next.js instead?

Comment (flybayer) [reply nextjs/run-tests.js:69]: this is fine

**Comment by flybayer** [comment]
@Skn0tt we need to update docs for this change?

**Comment by Skn0tt** [comment]
Don't think so, the docs don't mention any restrictions in terms of data types.

**Comment by Skn0tt** [comment]
The CI failure seems to be unrelated.

**Comment by flybayer** [comment]
@Skn0tt I just pushed a commit that cleans up your prefetch test a bit and adds /prefetching to the test precompile list.

But it still fails the test. Running it directly outside the test fails too

**Comment by Skn0tt** [comment]
8a17785 fixed it.

**Pull request merged**